### PR TITLE
NDI-15: Separated help box from panel on Export Event Participants Page

### PR DESCRIPTION
**Pull request by mihaelmladenov**
# Description:

### This PR styles:
* Export Event Participants page - it separates the help box from the main content of the page
* Visualizes the wizard bar globally
* Import Contacts page - adds specific styling for the wizard bar on this page
* Adjusts spacing around form buttons globally
* Fixes the name of a CSS file to match its function

# Screenshots:

## Export Event Participants page

**Before:**
before

**After:**
after

## Import Contributions page

**Before:**
33019222-18049184-ce02-11e7-81d0-9f00a9efc3df

**After:**
33019230-1fb6c29e-ce02-11e7-8350-ee6a4013c8a4

## Import Memberships page

**Before:**
33019172-e70f4952-ce01-11e7-94bc-83ae835325b4

**After:**
33025653-3cfa681e-ce17-11e7-84f8-f8ce9eaa2bf0

## Import Participants page

**Before:**
33019081-a04d0856-ce01-11e7-8af6-57541eb04e9c

**After:**
33019087-a67db3c4-ce01-11e7-8864-852445a6c2d9

### Step 1:

**Before:**
33018931-f50c0f64-ce00-11e7-8302-8dfafd5f4bae

**After:**
33025715-5abd07c6-ce17-11e7-8719-30e3b94b3191

### Step 2:

**Before:**
1344382581_import_contact_-_step_2_0_body_0_desktop

**After:**
1344382581_import_contact_-_step_2_0_body_0_desktop

### Step 3:

**Before:**
1344382581_import_contact_-_step_3_0_body_0_desktop

**After:**
1344382581_import_contact_-_step_3_0_body_0_desktop

### Step 4:

**Before:**
33018935-f9f1ca1e-ce00-11e7-9a61-a55c669b7800

**After:**
33018954-086a884c-ce01-11e7-8a02-790c14e8af8f

# Technical Details:
This PR changes the following files:

### `scss/civicrm/_variables.scss`:
* added a variable for the box-shadow that is used for elements inside a .crm-block that need a background color and shadow individually due to the fact that we have a help box or another element that should visually be outside of the white panel (i.e. we cannot apply the background color and shadow to the whole .crm-block div)

### `scss/civicrm/common/_buttons.scss`:
* removes left margin margin of form submit buttons and adds a default right margin

### `scss/civicrm/common/_common.scss`:
* includes the wizard styling file so that the wizard bar is visualized and styled

### `scss/civicrm/contact/pages/_export-contact.scss`:

* this file is renamed to `_export.scss` as it contains styling for all export pages, not just for the contact one
* removes background color and shadow for the whole content panel on the page to separate the help box above
* introduces a background color and shadow to the different elements below the help box
* changes margins to paddings on those elements to visually accomplish the desired look & feel
* visualizes the wizard bar

### `scss/civicrm/contact/pages/_import-contact.scss`:
* adjusts spacing around the wizard bar so that the wizard bar is exactly within the white panel
* adjusted left padding of table label cells

### `scss/civicrm/contact/pages/_member-import.scss`:
* adjusted spacing around form submit buttons
* adjusted left padding of table label cells

# Notes:
* BrowserStack tested
* Backstop.js tested

### CODE REVIEW

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 1]
```diff
@@ -1,9 +1,15 @@
 .CRM_Export_Form_Select {
+  padding-bottom: 5px;
+
   h2 {
-    padding-left: 20px;
+    background-color: #ffffff;
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export-contact.scss]: @mihaelmladenov Please use variables here

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 2]
```diff
@@ -27,6 +33,8 @@
   }
 
   #export-type {
+    background-color: #ffffff;
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export-contact.scss]: Variable Please

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 3]
```diff
@@ -36,10 +44,30 @@
...
-  .crm-submit-buttons {
-    > .crm-button {
-      margin-left: 0 !important;
-      margin-right: 10px !important;
+  .crm-form-block {
+    background-color: transparent;
+    box-shadow: none;
+
+    > .crm-submit-buttons {
+      background-color: #ffffff;
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export-contact.scss]: variable

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 4]
```diff
@@ -36,10 +44,30 @@
...
+      box-shadow: 0 8px 18px 0px rgba(0, 0, 0, .1);
+      margin: 0 !important;
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export-contact.scss]: What this much importants? can it be handled by adding more selectors or making style more specific?

**Comment by mukeshcompucorp** [comment]
We have such help on almost all pages why we are separating on this?
Can you confirm this with jon if this can be done other way as well or we really want to get that section out.

File: scss/civicrm/contact/pages/_export-contact.scss (hunk 2 above)
Comment (mukeshcompucorp) [reply scss/civicrm/contact/pages/_export-contact.scss]: does this have class? to handle css?

File: scss/civicrm/contact/pages/_export-contact.scss (hunk 3 above)
Comment (mukeshcompucorp) [reply scss/civicrm/contact/pages/_export-contact.scss]: Why we need it so say background white? isn't it white already?
Note that this component is already created for common solution. Why this needs a different solution then normal one? 
and if this solution is appropriate why not to apply at other places as well?

**Comment by mukeshcompucorp** [comment]
Instead of **Export Options (step 2 of 3)** the page should be showing wizard steps which are already designed notice the code `<ul class="wizard-bar">`.

It was hidden because it was not designed initially. Now that we are having it designed it should be visible.
@jonscreat please confirm this

**Comment by jamienovick** [comment]
@mukeshcompucorp we dont need to do wizard steps unless its super easy as this would be more work in this case. We don't need to change existing functionality of existing screens just to be clear. We just need to make them look nice and sensible in Shoreditch look and feel. thanks

**Comment by mukeshcompucorp** [comment]
@jamienovick we are not creating it. it is already there in page. It was us who hid intentionally as it was not looking so good. but, now that we have that component styled and we just need to remove `display: none;` I believe that should be done

**Comment by mihaelmladenov** [comment]
@mukeshcompucorp 
All comments have been solved. Please review again.

**Comment by mihaelmladenov** [comment]
@mukeshcompucorp I have run backstop tests and here are the results:
[html_report.zip](https://github.com/civicrm/org.civicrm.shoreditch/files/1482709/html_report.zip)

Can you please advise if this 1px left margin of buttons is an issue as well as whether to visualize the wizard steps on all pages that now have a hidden title since I imported the wizard stylesheet, as advised by you.

**Comment by mukeshcompucorp** [comment]
@mihaelmladenov 
You do not need to give me test results if passing.
In case of failure, we check if the change is for good or is messing our design.
If it looks good, you can take final confirmation from Jon about the screens which got change due to your style change if you can not determine it's ok to have or not.

@JonScreat can we add any common parts of CiviCRM (which you may already be adding) on,
how it should be designed,
like, a form design, a button panel etc. 
and can be added in our Wiki and Toolkit? This will be helpful for all FEs to determine how a page should look or what they can do to make it nearest to it if not 100%.

@mihaelmladenov keep this conversation in Shoreditch UI/UX room so, it a point is worth we can add it in Wiki directly after discussion and others can give their view points

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 5]
```diff
@@ -1,9 +1,15 @@
 .CRM_Export_Form_Select {
+  padding-bottom: 5px;
+
   h2 {
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export-contact.scss]: Why to style this when we have it hidden on page?

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 6]
```diff
@@ -36,10 +53,15 @@
...
-  .crm-submit-buttons {
-    > .crm-button {
-      margin-left: 0 !important;
-      margin-right: 10px !important;
+  .crm-form-block {
+    background-color: transparent;
+    box-shadow: none;
+
+    > .crm-submit-buttons {
+      box-shadow: 0 8px 18px 0px rgba(0, 0, 0, .1);
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export-contact.scss]: It seems that we are using this shadow on multiple places can this be converted to variable?

File: scss/civicrm/contact/pages/_import-contact.scss [hunk 7]
```diff
@@ -94,6 +94,11 @@
       padding: 20px 0 0 0 !important;
     }
 
+    #wizard-steps {
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_import-contact.scss]: Why there are 2 `#wizard-steps` in file?

Comment (mihaelmladenov) [reply scss/civicrm/contact/pages/_import-contact.scss]: The first one is for all the import contact pages - i.e. all the four steps. The second one is specifically for the Preview step, which is styled in a different way (i.e. there is no padding on the crm-block div).

Comment (mukeshcompucorp) [reply scss/civicrm/contact/pages/_import-contact.scss]: @mihaelmladenov can you add all screenshots?

Comment (mihaelmladenov) [reply scss/civicrm/contact/pages/_import-contact.scss]: @mukeshcompucorp I have added screenshots of all affected pages.

**Comment by mihaelmladenov** [comment]
@mukeshcompucorp Can you please review this again? The crm buttons horizontal spacing as well as the left margin in the tables have been fixed.

File: scss/civicrm/common/_common.scss [hunk 8]
```diff
@@ -19,4 +19,5 @@
 @import 'help';
 @import 'pagination';
 @import 'notifications';
+@import 'wizard';
```
Comment (AkA84) [inline scss/civicrm/common/_common.scss]: you are not creating this partial, so how come it wasn't in `common.scss` already? where was it used?

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 9]
```diff
@@ -1,9 +1,8 @@
 .CRM_Export_Form_Select {
```
Comment (AkA84) [inline scss/civicrm/contact/pages/_export-contact.scss]: this selector doesn't seem related specifically to "export contact", it has a pretty generic name. Which is it then? is it only for export contact or is it for any "export XYZ" page? if it's the latter, then any style contained in it shouldn't be in a partial called `_export-contact.scss`

File: scss/civicrm/_variables.scss [hunk 10]
```diff
@@ -15,6 +15,7 @@ $crm-form-layout-shadow: 0 3px 18px 0 rgba(48, 40, 40, 0.25);
 $crm-input-shadow: 0 0 3px 0 rgba(0, 0, 0, 0.2) inset;
 $crm-search-input-shadow: 0 1px 1px rgba(0, 0, 0, 0.075) inset;
 $crm-tab-shadow: 0 9px 9px 0 rgba(48, 40, 40, 0.25);
+$crm-panel-element-shadow: 0 3px 18px 0px rgba(0, 0, 0, .1);
```
Comment (AkA84) [inline scss/civicrm/_variables.scss]: why do we have so many different shadows?

File: scss/civicrm/contact/pages/_import-contact.scss [hunk 11]
```diff
@@ -21,7 +21,7 @@
   }
 
   #wizard-steps {
-    display: none;
+    margin: -30px -20px 0 -20px;
```
Comment (AkA84) [inline scss/civicrm/contact/pages/_import-contact.scss]: leave a comment explaining what this is for, it's a very weird style

File: scss/civicrm/contact/pages/_import-contact.scss (hunk 7 above)
Comment (AkA84) [reply scss/civicrm/contact/pages/_import-contact.scss]: why does it need to be nested inside the first `#wizard-steps` then? can't it be
```scss
#wizard-steps {

}

#Preview {

  #wizard-steps {

  }
}
```
this would make the selector much less confusing

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 12]
```diff
@@ -26,8 +25,19 @@
     margin-left: 0 !important;
   }
 
-  #export-type {
-    padding: 0 20px;
+  .crm-section {
+    margin-bottom: 0;
+    padding-bottom: 1em;
```
Comment (AkA84) [inline scss/civicrm/contact/pages/_export-contact.scss]: why `em`? we're using `px` basically everywhere

File: scss/civicrm/contact/pages/_export-contact.scss [hunk 13]
```diff
@@ -26,8 +25,19 @@
...
+
+    &.crm-export-mapping-section {
+      padding-bottom: 0;
+    }
+  }
+
+  .crm-exportOption-section {
+    background-color: $crm-white;
+    box-shadow: $crm-panel-element-shadow;
+    padding: 1em 20px;
```
Comment (AkA84) [inline scss/civicrm/contact/pages/_export-contact.scss]: why the mixed units?

File: scss/civicrm/common/_common.scss (hunk 8 above)
Comment (mihaelmladenov) [reply scss/civicrm/common/_common.scss]: The wizard itself was styled by Mukesh but he did not include it in this file. As I needed to work with the wizard, we agreed with Mukesh that I shall import this file and commit it.

File: scss/civicrm/contact/pages/_export-contact.scss (hunk 9 above)
Comment (mihaelmladenov) [reply scss/civicrm/contact/pages/_export-contact.scss]: I have renamed the file to `_export.scss` because all styling in it is for all export pages, not just for contacts.

File: scss/civicrm/contact/pages/_import-contact.scss (hunk 11 above)
Comment (mihaelmladenov) [reply scss/civicrm/contact/pages/_import-contact.scss]: we have the following HTML structure:

<div class="crm-block">
  <div id="wizard-steps">...</div>
  <h2>...</h2>
  <div class="help">...</div>
  ...other <div>'s with completely different classes
</div>

We have a white background color and padding on .crm-block so I am using a negative margin for the wizard steps in order to position it to match the .crm-block top, right and left borders.

File: scss/civicrm/contact/pages/_import-contact.scss (hunk 7 above)
Comment (mihaelmladenov) [reply scss/civicrm/contact/pages/_import-contact.scss]: The import wizard has several steps. One of the steps is Previewing. On this step we have a different HTML than the other ones, which means that we need a different left and right margin than the others. That's why I have done it this way.

File: scss/civicrm/_variables.scss (hunk 10 above)
Comment (mihaelmladenov) [reply scss/civicrm/_variables.scss]: The $crm-panel-element-shadow I am using because we have several elements right one after the other and each one needs to have a background color and a shadow due to the lack of a common container element for all these elements. This shadow visually looks like the design, so that's why I have added it.

File: scss/civicrm/common/_common.scss (hunk 8 above)
Comment (mukeshcompucorp) [reply scss/civicrm/common/_common.scss]: This wizard was styled as part of CiviHR and was forgotten to be added in this file when merging. We are just adding it here again. @mihaelmladenov your latest pull to develop should have added this already though.

File: scss/civicrm/contact/pages/_export.scss [hunk 14]
```diff
@@ -1,6 +1,9 @@
 .CRM_Export_Form_Select {
-  h2 {
-    padding-left: 20px;
+  padding-bottom: 5px;
+
+  .help {
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export.scss]: https://github.com/civicrm/org.civicrm.shoreditch/blob/develop/scss/civicrm/common/_help.scss
This file styles help section, do you need only margin change?

File: scss/civicrm/contact/pages/_export.scss [hunk 15]
```diff
@@ -22,8 +25,19 @@
...
+  .crm-section {
+    margin-bottom: 0;
+    padding-bottom: 20px;
+
+    &.crm-export-mapping-section {
+      padding-bottom: 0;
+    }
+  }
+
+  .crm-section {
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export.scss]: Can this styles be included at https://github.com/civicrm/org.civicrm.shoreditch/pull/101/files#diff-3a3f482f6902e2b4a41682a5288fb85bR28 ?

Comment (mukeshcompucorp) [reply scss/civicrm/contact/pages/_export.scss]: Line 28 same file

File: scss/civicrm/contact/pages/_export.scss [hunk 16]
```diff
@@ -32,10 +46,15 @@
...
 
-  .crm-submit-buttons {
-    > .crm-button {
-      margin-left: 0 !important;
-      margin-right: 10px !important;
+  .crm-form-block {
+    background-color: transparent;
+    box-shadow: none;
+
+    > .crm-submit-buttons {
```
Comment (mukeshcompucorp) [inline scss/civicrm/contact/pages/_export.scss]: Once, we style these commonly, it should not be here except here

File: scss/civicrm/contact/pages/_import-contact.scss (hunk 11 above)
Comment (mukeshcompucorp) [reply scss/civicrm/contact/pages/_import-contact.scss]: @mihaelmladenov I would need to understand this before we proceed on this margin

File: scss/civicrm/_variables.scss (hunk 10 above)
Comment (mukeshcompucorp) [reply scss/civicrm/_variables.scss]: Here we can use variable $box-shadow if we need to add shadow, or this is different?
